agent/eni/networkutils/utils_linux.go

```diff
@@ -17,22 +17,113 @@
 package networkutils
 
 import (
+	"context"
 	"path/filepath"
 	"strings"
+	"time"
 
 	"github.com/aws/amazon-ecs-agent/agent/eni/netlinkwrapper"
+	"github.com/aws/amazon-ecs-agent/agent/utils"
+	"github.com/pkg/errors"
 
-	log "github.com/cihub/seelog"
+	"github.com/cihub/seelog"
 )
 
+const (
+	// macAddressBackoffMin specifies the mimimum duration for the backoff
+	// when looking for an ENI's mac address on the host
+	macAddressBackoffMin = 2 * time.Millisecond
+
+	// macAddressBackoffMax specifies the maximum duration for the backoff
+	// when looking for an ENI's mac address on the host
+	macAddressBackoffMax = 200 * time.Millisecond
+
+	// macAddressBackoffJitter specifies the jitter multiple percentage when
+	// looking for an ENI's mac address on the host
+	macAddressBackoffJitter = 0.2
```

Comment:
I don't think jitter is really necessary here since you're not making remote calls or accessing a resource with high concurrency.  On the other hand, I don't think it really hurts either except to make the timing a bit more unpredictable.